app/validation.go

```diff
@@ -6,109 +6,198 @@ import (
 
 	"code.cloudfoundry.org/fissile/model"
 	"code.cloudfoundry.org/fissile/validation"
+
+	yaml "gopkg.in/yaml.v2"
 )
 
 // Validate applies a series of checks to the role
 // manifest and opinions, testing for consistency against each other
 // and the loaded bosh releases. The result is a (possibly empty)
 // array of any issues found.
 func (f *Fissile) Validate() validation.ErrorList {
+	errors := make(chan *validation.Error)
+	validator, err := newValidator(f, errors)
+	if err != nil {
+		return validation.ErrorList{err}
+	}
+	go validator.validate()
 	allErrs := validation.ErrorList{}
+	for err := range errors {
+		allErrs = append(allErrs, err)
+	}
+	return allErrs
+}
 
+type validator struct {
+	errOut           chan<- *validation.Error
+	f                *Fissile
+	propertyDefaults model.PropertyDefaults
+	lightOpinions    map[string]string
+	darkOpinions     map[string]string
+	variableUsage    map[string]int
+}
+
+func newValidator(f *Fissile, errOut chan<- *validation.Error) (*validator, *validation.Error) {
 	opinions, err := model.NewOpinions(f.Options.LightOpinions, f.Options.DarkOpinions)
 	if err != nil {
-		return append(allErrs, validation.GeneralError("Light are dark opinions could not be read", err))
+		return nil, validation.GeneralError("Light are dark opinions could not be read", err)
 	}
 
-	boshPropertyDefaultsAndJobs := f.collectPropertyDefaults()
-	darkOpinions := model.FlattenOpinions(opinions.Dark, false)
-	lightOpinions := model.FlattenOpinions(opinions.Light, false)
-	manifestProperties := collectManifestProperties(f.Manifest)
+	return &validator{
+		errOut:           errOut,
+		f:                f,
+		propertyDefaults: f.collectPropertyDefaults(),
+		lightOpinions:    model.FlattenOpinions(opinions.Light, false),
+		darkOpinions:     model.FlattenOpinions(opinions.Dark, false),
+		variableUsage:    make(map[string]int),
+	}, nil
+}
+
+func (v *validator) validate() {
+	defer close(v.errOut)
+
+	v.checkForSortedProperties(
+		"configuration.templates",
+		v.f.Manifest.Configuration.RawTemplates)
 
-	// All properties must be defined in a BOSH release
-	allErrs = append(allErrs, checkForUndefinedBOSHProperties("role-manifest",
-		manifestProperties, boshPropertyDefaultsAndJobs)...)
+	for propertyName, templateDef := range v.f.Manifest.Configuration.Templates {
+		if templateDef.IsGlobal {
+			v.checkForUndefinedProperty(
+				"configuration.templates",
+				propertyName,
+				v.propertyDefaults)
```

Comment:
This is the weird part.

Reply:
This is because we call `v.checkForUndefinedProperty` elsewhere, with the instance-group-specific properties instead. I believe this is the only validation function we do this.

Reply:
Fixed in a7c161c2d8fa267f03009f7b9f063996b53c8e4b